NickvisionTagger.Shared/Models/MusicFile.cs

```diff
@@ -527,127 +525,189 @@ public bool RemoveCustomProperty(string name)
+            return false;
         }
-        else if (formatString == "%title%- %artist%")
+        var matches = Regex.Matches(formatString, @"%(\w+)%", RegexOptions.IgnoreCase); //wrapped in %%
+        if(matches.Count == 0)
         {
-            var dashIndex = Filename.IndexOf("- ");
-            if(dashIndex == -1)
+            return false;
+        }
+        var splits = Regex.Split(formatString, @"(\%\w+\%)", RegexOptions.IgnoreCase).Where(x =>
+        {
+            if(x.Length > 1)
             {
-                return false;
+                return x[0] != '%' && x[x.Length - 1] != '%';
             }
-            Title = Filename.Substring(0, dashIndex);
-            Artist = Filename.Substring(dashIndex + 2, Filename.IndexOf(System.IO.Path.GetExtension(Path)) - (Title.Length + 2));
-        }
-        else if (formatString == "%track%- %title%")
+            return x.Length != 0;
+        }).ToList();
+        foreach(var s in splits)
         {
-            var dashIndex = Filename.IndexOf("- ");
-            if(dashIndex == -1)
+            if(!Filename.Contains(s))
             {
                 return false;
             }
+        }
+        var filename = System.IO.Path.GetFileNameWithoutExtension(Filename);
+        var i = 0;
+        foreach(Match match in matches)
+        {
+            string value = match.Value.Remove(0, 1); //remove first %
+            value = value.Remove(value.Length - 1, 1).ToLower(); //remove last %;
+            var len = 0;
             try
             {
-                var trackString = Filename.Substring(0, dashIndex);
-                Track = int.Parse(trackString);
-                Title = Filename.Substring(dashIndex + 2, Filename.IndexOf(System.IO.Path.GetExtension(Path)) - (trackString.Length + 2));
+                len = filename.IndexOf(splits[i]);
             }
             catch
             {
-                Track = 0;
-                Title = "";
+                len = filename.Length;
             }
-        }
-        else if (formatString == "%title%")
-        {
-            Title = Filename.Substring(0, Filename.IndexOf(System.IO.Path.GetExtension(Path)));
-        }
-        else
-        {
-            return false;
-        }
-        return true;
-    }
-    
-    /// <summary>
-    /// Uses the music file's tag to set the file's filename in the format of the format string
-    /// </summary>
-    /// <param name="formatString">Th format string</param>
-    /// <returns>True if successful, else false</returns>
-    public bool TagToFilename(string formatString)
-    {
-        if (formatString == "%artist%- %title%")
-        {
-            if(string.IsNullOrEmpty(Artist) || string.IsNullOrEmpty(Title))
+            if(value == "title" || value == _("title"))
             {
-                return false;
+                Title = filename.Substring(0, len);
```

Comment:
Perhaps don't duplicate `filename.Substring(0, len)` 24 times :sweat_smile:

Reply:
I can store it as
`var str = filename.Substring(0, len)` and then duplicate `str` 24 times....to me it's the same
either way tho, the code will only be called once out of the 24 times depending on `value` so no performance issue here...